Framework/PythonInterface/test/python/plugins/algorithms/SANSDarkRunBackgroundCorrectionTest.py

```diff
@@ -0,0 +1,763 @@
+
+    def _check_output_workspace(self, original_ws, corrected_ws, expected_correction_value):
+        # Iterate over all spectra
+        num_spectra = original_ws.getNumberHistograms()
+
+        for index in range(0, num_spectra):
+            y_original = original_ws.dataY(index)
+            y_corrected = corrected_ws.dataY(index)
+            for elem in range(0, len(y_original)):
+                expected = y_original[elem] - expected_correction_value
+                self.assertAlmostEqual(expected,
+                                       y_corrected[elem], 4)
+
+    def _do_run_dark_subtraction(self, scatter, dark_run, mean, uniform, normalization_ratio,
+                                 out_ws_name, applyToMonitors, applyToDetectors, selected_monitor):
+        alg_dark  = AlgorithmManager.createUnmanaged("SANSDarkRunBackgroundCorrection")
+        alg_dark.initialize()
+        alg_dark.setChild(True)
+        alg_dark.setProperty("InputWorkspace", scatter)
+        alg_dark.setProperty("DarkRun", dark_run)
+        alg_dark.setProperty("Mean", mean)
+        alg_dark.setProperty("Uniform", uniform)
+        alg_dark.setProperty("NormalizationRatio", normalization_ratio)
+        alg_dark.setProperty("OutputWorkspace", out_ws_name)
+        alg_dark.setProperty("ApplyToMonitors", applyToMonitors)
+        alg_dark.setProperty("ApplyToDetectors", applyToDetectors)
+        alg_dark.setProperty("SelectedMonitors", selected_monitor)
+        alg_dark.execute()
+
+        return alg_dark.getProperty("OutputWorkspace").value
+
+    def _check_output_workspace_non_uniform(self, original_ws, corrected_ws,
+                                            dark_ws, expected_correction_value):
+        # Iterate over all spectra
+        num_spectra = original_ws.getNumberHistograms()
+
+        for index in range(0, num_spectra):
+            y_original = original_ws.dataY(index)
+            y_dark = dark_ws.dataY(index)
+            y_corrected = corrected_ws.dataY(index)
+            for elem in range(0, len(y_original)):
+                expected = y_original[elem] - y_dark[elem]*expected_correction_value
+                self.assertAlmostEqual(expected, y_corrected[elem], 4)
+
+    def _provide_workspace2D(self, bin_boundaries, y_value, e_value, name, spectra, use_y_list = False):
+        x = spectra*[element for element in xrange(bin_boundaries)]
+        y = None
+        if use_y_list:
+            y = y_value
+        else:
+            y = spectra*[y_value for element in xrange(bin_boundaries - 1)]
+        e = spectra*[e_value for element in xrange(bin_boundaries - 1)]
+        self._create_test_workspace(name, x, y, e, spectra)
+
+    def _clean_up(self, ws_to_clean):
+        for ws in ws_to_clean:
+            if AnalysisDataService.doesExist(ws):
+                AnalysisDataService.remove(ws)
+
+    def _load_workspace_with_monitors(self, monY, monE, dataY, dataE, as_dark_run = False):
```

Comment:
This is getting called a lot! 
1. Do you really need to get test data from disk?
2. Wouldn't it be better to load and cache the data once?